include/neml2/dispatchers/WorkDispatcher.h

```diff
@@ -53,8 +59,7 @@ namespace neml2
  * general, each work dispatch involves four steps:
  * 1. Work generation: The work generator generates the next \p n batches of work.
  * 2. Preprocessing: The dispatcher preprocesses the work.
- * 3. Do work: The dispatcher dispatches the work to a worker, and the worker completes and returns
- *    the result.
+ * 3. Do work: The worker performs the work.
  * 4. Postprocessing: The dispatcher postprocesses the result.
  *
```

Comment:
Let's add a brief summary of the threading model and its current limitations here (i.e. right now each device has its own thread).  Probably should leave some warning to the future about the stupid jit issues as well.